ts/features/idpay/configuration/screens/IdPayDiscountInstrumentsScreen.tsx

```diff
@@ -0,0 +1,134 @@
+import { IDPayConfigurationParamsList } from "../navigation/navigator";
+import TopScreenComponent from "../../../../components/screens/TopScreenComponent";
+import { useIdPayInfoCieBottomSheet } from "../../code/components/IdPayInfoCieBottomSheet";
+
+type IdPayDiscountInstrumentsScreenRouteParams = {
+  initiative?: InitiativeDTO;
+};
+
+type IdPayDiscountInstrumentsScreenRouteProps = RouteProp<
+  IDPayConfigurationParamsList,
+  "IDPAY_CONFIGURATION_DISCOUNT_INSTRUMENTS"
+>;
+
+const IdPayDiscountInstrumentsScreen = () => {
+  const dispatch = useIODispatch();
+  const route = useRoute<IdPayDiscountInstrumentsScreenRouteProps>();
+  const navigation =
+    useNavigation<IOStackNavigationProp<IDPayConfigurationParamsList>>();
+  const { initiative } = route.params;
+
+  const initiativePaymentMethods = useIOSelector(
+    idpayDiscountInitiativeInstrumentsSelector
+  );
+  const isLoadingPaymentMethods = useIOSelector(
+    isLoadingDiscountInitiativeInstrumentsSelector
+  );
+
+  const { bottomSheet, present: presentCIEBottomSheet } =
+    useIdPayInfoCieBottomSheet();
+
+  React.useEffect(() => {
+    if (initiative) {
+      dispatch(
+        idpayDiscountInitiativeInstrumentsGet.request({
+          initiativeId: initiative.initiativeId
+        })
+      );
+    }
+  }, [initiative, dispatch]);
+
+  const handleBackPress = () => navigation.goBack();
+
+  const handlePaymentMethodValueChange = (
+    paymentMethodType: IDPayDiscountInitiativeInstruments,
+    value: boolean
+  ) => {
+    // if (value) {
+    //   navigation.navigate(
+    //     IDPayConfigurationParamsList.IDPAY_CONFIGURATION_INSTRUMENTS_ENROLLMENT,
+    //     {
+    //       initiative,
+    //       paymentMethodType
+    //     }
+    //   );
+    // }
+    // console.log(paymentMethodType, value);
+  };
+
+  const handlePressActionButton = (
+    paymentMethodType: IDPayDiscountInitiativeInstruments
+  ) => {
+    if (paymentMethodType === IDPayDiscountInitiativeInstruments.CIE) {
+      presentCIEBottomSheet();
+    }
+  };
+
+  return (
+    <>
+      <TopScreenComponent
+        goBack={handleBackPress}
+        contextualHelp={emptyContextualHelp}
+      >
+        <LoadingSpinnerOverlay
+          isLoading={isLoadingPaymentMethods}
+          loadingOpacity={1}
+        >
+          <ScrollView style={styles.container}>
+            <H1>
+              {I18n.t("idpay.configuration.instruments.paymentMethods.header")}
+            </H1>
+            <VSpacer size={8} />
+            <Body>
+              {I18n.t("idpay.configuration.instruments.paymentMethods.body", {
+                initiativeName: initiative?.initiativeName ?? ""
+              })}
+            </Body>
+            <VSpacer size={24} />
+            {initiativePaymentMethods.map(paymentMethod => (
+              <IdPayDiscountInstrumentEnrollmentSwitch
+                key={paymentMethod.idWallet}
```

Comment:
`idWallet` could be `undefined`. `instrumentId` is always defined and unique.

```suggestion
                key={paymentMethod.instrumentId}
```